tests/test_torch_agent.py

```diff
@@ -897,14 +864,51 @@ def test_interactive_mode(self):
             'Evaluating 0', response['text'], 'Incorrect output in single act()'
         )
 
-        # finally, the expected failure
-        with self.assertRaises(RuntimeError):
-            agent = get_agent(batchsize=16, interactive_mode=False)
-            agent.observe(Message({'text': 'foo', 'episode_done': True}))
-            response = agent.act()
-            shared = create_agent_from_shared(agent.share())
-            shared.observe(Message({'text': 'bar', 'episode_done': True}))
-            response = shared.act()
+        # finally, actively attempt to sabotage
+        agent = get_agent(batchsize=16, interactive_mode=False)
+        agent.observe(Message({'text': 'foo', 'episode_done': True}))
+        response = agent.act()
+        self.assertIn(
+            'Evaluating 0', response['text'], 'Incorrect output in single act()'
+        )
+        shared = create_agent_from_shared(agent.share())
+        shared.observe(Message({'text': 'bar', 'episode_done': True}))
+        response = shared.act()
+        self.assertIn(
+            'Evaluating 0', response['text'], 'Incorrect output in single act()'
+        )
+
+    def test_use_reply(self):
+        """Check that self-observe is correctly acting on labels."""
+        # default is hybrid label-model, which uses the label if it's available, and
+        # otherwise the label
```

Comment:
`uses the label if it's available, and otherwise the label` what does this mean?